GTG/core/saved_searches.py

```diff
@@ -0,0 +1,133 @@
+# -----------------------------------------------------------------------------
+# Getting Things GNOME! - a personal organizer for the GNOME desktop
+# Copyright (c) The GTG Team
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+from uuid import uuid4
+from dataclasses import dataclass, field
+import logging
+
+from lxml.etree import Element, SubElement
+from typing import List
+
+from GTG.core.base_store import BaseStore
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class SavedSearch:
+    """A saved search."""
+
+    id: uuid4
+    name: str
+    query: str
+    icon: str = None
+    children: List = field(default_factory=list)
+
+    def __str__(self) -> str:
+        """String representation."""
+
+        return (f'Saved Search "{self.name}" '
+                f'with query "{self.query}" and id "{self.id}"')
```

Comment:
IMO it should be `self.name`, and use `__repr__` for the more detailed info

Reply:
Agree